airflow/providers/google/cloud/sensors/pubsub.py

```diff
@@ -170,22 +171,41 @@ def execute(self, context: Context) -> None:
                     subscription=self.subscription,
                     max_messages=self.max_messages,
                     ack_messages=self.ack_messages,
-                    messages_callback=self.messages_callback,
                     poke_interval=self.poke_interval,
                     gcp_conn_id=self.gcp_conn_id,
                     impersonation_chain=self.impersonation_chain,
                 ),
                 method_name="execute_complete",
             )
 
-    def execute_complete(self, context: dict[str, Any], event: dict[str, str | list[str]]) -> str | list[str]:
+    def execute_complete(self, context: Context, event: dict[str, str | list[str]]) -> Any:
         """Return immediately and relies on trigger to throw a success event. Callback for the trigger."""
         if event["status"] == "success":
             self.log.info("Sensor pulls messages: %s", event["message"])
+            if self.messages_callback:
+                received_message_fmt = self._convert_to_received_message(event["message"])
```

Comment:
May I know why can't we just reuse the logic [here](https://github.com/apache/airflow/blob/111c6327bb5c2b4dd4421c8557b8b7de670a01e3/airflow/providers/google/cloud/sensors/pubsub.py#L148C1-L148C83)?

Reply:
We need to convert back to the original pubsub response type to work with user `messages_callback`, the event received from the trigger is list[dict].

Reply:
Also if no` message_callback` is provided , we can return immediately because the event message is already converted to list of dict in triggerer.